web/yo/app/scripts/controllers/gene.js

```diff
@@ -1538,40 +1538,144 @@ angular.module('oncokbApp')
+                } else if (mutation) {
+                    result[0] = $scope.gene.mutations.indexOf(mutation);
+                }
+                return result;
+            }
+
+            $scope.initGeneStatus = function(mutation, tumor, ti, treatment) {
+                var objects = [mutation, tumor, ti, treatment];
+                var indices = getIndex(mutation, tumor, ti, treatment);
+                $scope.geneStatus = loopInitGeneStatus(objects, $scope.geneStatus, indices, 0);
+            };
+
+            $scope.checkGeneStatus = function(mutation, tumor, ti, treatment, key, statusType) {
+                if (!_.isString(statusType)) {
+                    statusType = 'isOpen';
+                }
+                var indices = getIndex(mutation, tumor, ti, treatment);
+                var result = loopCheckGeneStatus(false, $scope.geneStatus, indices, 0, key, statusType);
+                return result;
+            };
+
+            $scope.getGeneStatusItem = function(mutation, tumor, ti, treatment, key) {
+                var indices = getIndex(mutation, tumor, ti, treatment);
+                var result = loopGetGeneStatusItem(null, $scope.geneStatus, indices, 0, key);
+                return result;
+            };
+
+            function loopCheckGeneStatus(status, geneStatus, indices, index, key, statusType) {
+                var strIndex = indices[index].toString();
+                if (geneStatus.hasOwnProperty(strIndex)) {
+                    var nextIndex = index + 1;
+                    if (!_.isNumber(indices[nextIndex]) || indices[nextIndex] === -1) {
+                        if (key) {
+                            return geneStatus[strIndex][key] ? geneStatus[strIndex][key][statusType] : false;
+                        }
+                        return geneStatus[strIndex][statusType];
+                    }
+                    status = loopCheckGeneStatus(status, geneStatus[strIndex], indices, ++index, key, statusType);
+                }
+                return status;
+            }
+
+            function loopGetGeneStatusItem(ref, geneStatus, indices, index, key) {
+                var strIndex = indices[index].toString();
+                if (geneStatus.hasOwnProperty(strIndex)) {
+                    var nextIndex = index + 1;
+                    if (!_.isNumber(indices[nextIndex]) || indices[nextIndex] === -1) {
+                        if (key) {
+                            return geneStatus[strIndex][key] ? geneStatus[strIndex][key] : null;
+                        }
+                        return geneStatus[strIndex];
+                    }
+                    ref = loopGetGeneStatusItem(ref, geneStatus[strIndex], indices, ++index, key);
+                }
+                return ref;
+            }
+
+            function loopInitGeneStatus(objects, geneStatus, indices, index) {
+                var defaultIsOpen = Boolean($scope.reviewMode);
```

Comment:
@JiaoJiao123 could you help me specifically check whether I still need to check the review mode here? And also the similar code below.